cli/ssh_test.go

```diff
@@ -644,32 +646,36 @@ func TestSSH(t *testing.T) {
 			"8222:"+httpServer.Listener.Addr().String(),
 		)
 		clitest.SetupConfig(t, client, root)
-		pty := ptytest.New(t).Attach(inv)
-		inv.Stderr = pty.Output()
 
 		ctx, cancel := context.WithTimeout(context.Background(), testutil.WaitLong)
 		defer cancel()
 
 		cmdDone := tGo(t, func() {
 			err := inv.WithContext(ctx).Run()
-			assert.NoError(t, err, "ssh command failed")
+			// fails because we cancel context to close
+			assert.Error(t, err, "ssh command should fail")
 		})
 
-		// Agent is still starting
-		pty.ExpectMatch("Waiting")
-
-		_ = agenttest.New(t, client.URL, agentToken)
-		coderdtest.AwaitWorkspaceAgents(t, client, workspace.ID)
-
-		// Startup script has just finished
-		pty.ExpectMatch(startupScriptPattern)
-
-		// Download the test page
-		pty.WriteLine("curl localhost:8222")
-		pty.ExpectMatch("hello world")
+		require.Eventually(t, func() bool {
+			req, err := http.NewRequestWithContext(ctx, http.MethodGet, "http://localhost:8222/", nil)
+			if !assert.NoError(t, err) {
+				// true exits the loop.
+				return true
+			}
+			resp, err := http.DefaultClient.Do(req)
```

Comment:
I wonder if this may leave idle connections after it (triggering leak detection). If it does though, perhaps we just need to handle it in test main so we can keep using the default client.

Reply:
I haven't seen any.  I think idle connections are cleaned up promptly.